File: DTOs/SpotDto.cs
using System.ComponentModel.DataAnnotations;
using SmartParking.Models;

namespace SmartParking.Models.DTOs
{
    public class SpotDto
    {
        [Required]
        public string Code { get; set; } = string.Empty;

        [Required]
        public int ZoneId { get; set; }

        public SpotStatus Status { get; set; } = SpotStatus.Free;
    }
}

File: frontend/smart-parking/src/app/models/spot.dto.ts
export interface UpdateSpotDto extends SpotDto {
  id: number;
}

export enum SpotStatus {
  Free     = 0,
  Occupied = 1,
  Reserved = 2,
}
 
/** What you POST/PUT to the API (no id) */
export interface SpotDto {
  code: string;
  zoneId: number;
  status: SpotStatus;
}
 
/** What the API returns (has id + zone/parking info) */
export interface Spot {
  id: number;
  code: string;
  zoneId: number;
  zoneName: string;
  parkingId: number;
  parkingName: string;
  status: SpotStatus;
  updating?: boolean; // local UI flag
}